identity/src/commonMain/kotlin/com/android/identity/util/ByteArrayUtil.kt

```diff
@@ -0,0 +1,230 @@
+package com.android.identity.util
+
+/**
+ * ByteArray extension functions for consistent reading and writing commonly used types.
+ */
+
+fun ByteArray.putInt8(offset: Int, value: Int) {
+    require(value in Byte.MIN_VALUE..Byte.MAX_VALUE) { "Value $value is out of Int8 range" }
```

Comment:
I think there are a number of places in the code where the valid range is less than the full possible range... I'm wondering if it'd be a helpful pattern to have each of these functions take an optional range argument for this check. If none is provided, we could default to this range.

fun ByteArray.putInt8(offset: Int, value: Int, validRange: IntRange = Byte.MIN_VALUE..Byte.MAX_VALUE) {
...